app/controllers/advertisement_clicks_controller.rb

```diff
@@ -21,30 +21,58 @@ def show
 
   private
 
+  def sponsor?
+    request.path.include? "/visit-sponsor"
+  end
+
   def set_variables
-    @campaign = Campaign.select(:id, :name, :url).find_by(id: params[:campaign_id])
-    @creative = Creative.select(:id, :name).find_by(id: params[:creative_id])
     @property = Property.select(:id, :name, :url).find_by(id: params[:property_id])
+
+    if sponsor?
+      @campaign = @property.current_sponsor_campaign
```

Comment:
If it's a fallback sponsor, do they link to our homepage?

Reply:
They get sent to the advertiser page if we hit the catch all.